src/plugins/intel_cpu/src/nodes/tensoriterator.cpp

```diff
@@ -54,12 +54,15 @@ static NodeConfig make_plain_config(const std::shared_ptr<ov::Node>& op) {
     return config;
 }
 
-static void redefineToMemories(const std::vector<MemoryPtr>& to_mems, MemoryDescPtr new_desc) {
+static void redefineToMemories(const std::vector<MemoryPtr>& to_mems, MemoryDescPtr new_desc, void* data = nullptr) {
     const auto &currDesc = to_mems.front()->getDesc();
     if (currDesc.getShape().isDynamic() || currDesc.getShape().getStaticDims() != new_desc->getShape().getStaticDims()) {
         // TODO : check the entire dstMemPtrs usage considering the proper memory sharing
         for (size_t j = 0; j < to_mems.size(); j++) {
-            to_mems[j]->redefineDesc(new_desc);
+            if (nullptr != data)
+                to_mems[j]->Create(new_desc, data, false);
```

Comment:
The motivation of this code change is to reset internal memory buffer with a new data buffer. M.b. a better way is to overload redefineDesc(desc). How do you think?

Reply:
No the problem is that `Creat` call completely resets the memory object, and, for example if it share the memory manager with some other objects, this connection will be lost. Also, which is more important, from the design perspective, the memory object, being a black box, may have any other important internal state, which may be rest by the `Creat` call. I think this is not what we want here.

Reply:
> No, the problem is that `Creat` call completely resets the memory object, and, for example if it share the memory manager with some other objects, this connection will be lost. Also, which is more important, from the design perspective, the memory object, being a black box, may have any other important internal state, which may be rest by the `Creat` call. I think this is not what we want here. BTW, if we want to reset data pointer, we use `setDataHandle` call. But this means that we switch the memory object to another externally controlled memory blob. Which in case of memory sharing breaks the connection with common mem manager.


Hence got a question here - "redefineDesc" basically is "Create". How could the original code work as it uses this method to reset "to" memory object?

Reply:
defer it as it contributes a little bit but asks much effort.